src/server/bg_services/log_replication_scanner.js

```diff
@@ -122,93 +122,79 @@ class LogReplicationScanner {
     async process_candidates(src_bucket, dst_bucket, candidates) {
         const src_dst_objects_list = await this.head_objects(src_bucket.name, dst_bucket.name, candidates);
 
+        const copy_keys = [];
+        const delete_keys = [];
+
         for (const candidate of src_dst_objects_list) {
-            await this.process_candidate(src_bucket, dst_bucket, candidate);
+            await this.process_candidate(copy_keys, delete_keys, src_bucket, dst_bucket, candidate);
         }
+
+        this.copy_objects(src_bucket.name, dst_bucket.name, copy_keys);
+        this.delete_objects(dst_bucket.name, delete_keys);
     }
 
-    async process_candidate(src_bucket, dst_bucket, candidate) {
+    async process_candidate(copy_keys, delete_keys, src_bucket, dst_bucket, candidate) {
```

Comment:
`process_candidate` goes over all occurrences of a specific key and performs the operation for it. But this is redundant, and I think it can also result in wrong deletions.
I suggest processing only the most recent item in `items` and ignoring the rest

Reply:
Agreed, I think we can store the most recent logs for the key in candidate itself.